api/src/main/java/org/openmrs/api/db/hibernate/PersonLuceneQuery.java

```diff
@@ -112,6 +117,26 @@ public PersonLuceneQuery(SessionFactory sessionFactory) {
 	 * @param gender the gender of the person to search  
 	 * @return the LuceneQuery that returns Persons with a soundex representation of the firstName and other defined search criteria
 	 */
+	public LuceneQuery<PersonName> getSoundexPersonNameSearchOnThreeNames(String n1, String n2, String n3,  Integer birthyear, boolean includeVoided, String gender) {
+		
+		String threeNameQuery =  THREE_NAME_QUERY.replace("n1", LuceneQuery.escapeQuery(n1))
+			.replace("n2", LuceneQuery.escapeQuery(n2))
+			.replace("n3", LuceneQuery.escapeQuery(n3));
+		
+		threeNameQuery = "(" +  threeNameQuery.replace("n1", n1).replace("n2", n2).replace("n3", n3) + ")";
```

Comment:
The calls to `replace()` here are redundant aren't they? I.e., we already did this substitution above.

Reply:
You are right